samza-core/src/main/java/org/apache/samza/system/SystemAdmins.java

```diff
@@ -38,6 +38,10 @@ public SystemAdmins(Config config) {
     this.systemAdminMap = systemConfig.getSystemAdmins();
   }
 
+  public SystemAdmins(Map<String, SystemAdmin> systemAdminMap) {
```

Comment:
Is this only for tests? Could you mock it instead in the tests? It would be good to avoid having public methods just for tests.

Reply:
Done.